src/keys/slip10.rs

```diff
@@ -172,6 +192,19 @@ pub mod secp256k1 {
         }
     }
 
+    impl Bip44IntoIter for secp256k1_ecdsa::SecretKey {
```

Comment:
Feels like this should be implemented on `Bip44` with a generic type, and use `&self`.

Reply:
Sure. It would be nice to have something like:

```rust
trait ToChain {
    type Chain<K>;
    fn to_chain<K>(&self) -> Self::Chain<K>;
}
```

But the impl is impossible without generic specializations:

```rust
impl ToChain for Bip44 {
    type Chain<ed25519::SecretKey> = [Hardened; 5];
    type Chain<secp256k1_ecdsa::SecretKey> = [u32; 5];
    fn to_chain<ed25519::SecretKey>(&self) -> Chain<ed25519::SecretKey>;
    fn to_chain<secp256k1_ecdsa::SecretKey>(&self) -> Chain<secp256k1_ecdsa::SecretKey>;
}
```

The best I can do is as follows:

```rust
pub trait ToChain<C>: Sealed {
    type Chain;
    fn to_chain(pre_chain: &C) -> Self::Chain;
}
```

and impl it for `ed25519::SecretKey` and `secp256k1_ecdsa::SecretKey`.

Reply:
Maybe you have better ideas.

Reply:
Why not just
```rust
pub trait ToChain<K>: Sealed {
    type Chain;
    fn to_chain(&self) -> Self::Chain;
}

impl ToChain<ed25519::SecretKey> for Bip44 {
    type Chain = [Hardened; 5];
    ...
}

impl ToChain<secp256k1_ecdsa::SecretKey> for Bip44 {
    type Chain = [u32; 5];
    ...
}
```

Reply:
It would be difficult to call `bip44.to_chain()` for a specific key type. Need to resolve for which key: `<Bip44 as ToChain<ed25519::SecretKey>>::to_chain(bip44)`.

Reply:
Yes, or it can be resolved via bounds. Another option is to have the traits implemented on the keys and have a fn defined for `Bip44` like so:

```rust
fn to_chain<K: ToChain>(&self) -> K::Chain {...}
```

Reply:
Updated. Added `Bip44::to_chain`.